From 98a54e735e3ceb056391da2560add760c4ec35c4 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Tue, 6 Jun 2023 09:40:03 +0200
Subject: [PATCH 1/4] Use wide chars explicitly to open the drive

---
 src/device.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/device.c b/src/device.c
index 34b610b..c0c14e7 100644
--- a/src/device.c
+++ b/src/device.c
@@ -476,7 +476,7 @@ static int libc_open ( dvdcss_t dvdcss, const char *psz_device )
 #if defined( _WIN32 )
 static int win2k_open ( dvdcss_t dvdcss, const char *psz_device )
 {
-    char psz_dvd[7] = "\\\\.\\\0:";
+    WCHAR psz_dvd[7] = L"\\\\.\\\0:";
     psz_dvd[4] = psz_device[0];
 
     /* To work around an M$ bug in IOCTL_DVD_READ_STRUCTURE, we need read
@@ -488,14 +488,14 @@ static int win2k_open ( dvdcss_t dvdcss, const char *psz_device )
      * (See Microsoft Q241374: Read and Write Access Required for SCSI
      * Pass Through Requests) */
     dvdcss->i_fd = (int)
-                CreateFile( psz_dvd, GENERIC_READ | GENERIC_WRITE,
+                CreateFileW( psz_dvd, GENERIC_READ | GENERIC_WRITE,
                             FILE_SHARE_READ | FILE_SHARE_WRITE,
                             NULL, OPEN_EXISTING,
                             FILE_FLAG_RANDOM_ACCESS, NULL );
 
     if( (HANDLE) dvdcss->i_fd == INVALID_HANDLE_VALUE )
         dvdcss->i_fd = (int)
-                    CreateFile( psz_dvd, GENERIC_READ, FILE_SHARE_READ,
+                    CreateFileW( psz_dvd, GENERIC_READ, FILE_SHARE_READ,
                                 NULL, OPEN_EXISTING,
                                 FILE_FLAG_RANDOM_ACCESS, NULL );
 
-- 
2.37.3.windows.1

