From 3a69753af76268f4e57ba50c05d13ca73d850e0b Mon Sep 17 00:00:00 2001
From: Fatih Uzunoglu <fuzun54@outlook.com>
Date: Wed, 21 Feb 2024 20:27:36 +0200
Subject: [PATCH] Use fxc2 through wine instead of fxc

---
 tools/qsb/qsb.cpp | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/tools/qsb/qsb.cpp b/tools/qsb/qsb.cpp
index 3700a07a89..b761bb3913 100644
--- a/tools/qsb/qsb.cpp
+++ b/tools/qsb/qsb.cpp
@@ -81,6 +81,7 @@ static bool runProcess(const QString &binary, const QStringList &arguments,
 {
 #if QT_CONFIG(process)
     QProcess p;
+    p.setWorkingDirectory(QCoreApplication::applicationDirPath());
     p.start(binary, arguments);
     const QString cmd = binary + QLatin1Char(' ') + arguments.join(QLatin1Char(' '));
     if (!silent)
@@ -832,15 +833,15 @@ int main(int argc, char **argv)
                                               QLatin1String("/nologo"),
                                               QLatin1String("/E"), QString::fromLocal8Bit(s.entryPoint()),
                                               QLatin1String("/T"), QString::fromLocal8Bit(typeArg),
-                                              QLatin1String("/Fo"), QDir::toNativeSeparators(tmpOut)
+                                              QLatin1String("/Fo"), "Z:" + QDir::toNativeSeparators(tmpOut)
                                           });
                     if (cmdLineParser.isSet(debugInfoOption))
                         arguments << QLatin1String("/Od") << QLatin1String("/Zi");
-                    arguments.append(QDir::toNativeSeparators(tmpIn));
+                    arguments.append("Z:" + QDir::toNativeSeparators(tmpIn));
                     QByteArray output;
                     QByteArray errorOutput;
-                    const QString compilerName = useDxc ? QLatin1String("dxc") : QLatin1String("fxc");
-                    bool success = runProcess(compilerName, arguments, &output, &errorOutput);
+                    const QString compilerName = useDxc ? QLatin1String("dxc") : QLatin1String("fxc2");
+                    bool success = runProcess(QLatin1String("wine"), arguments, &output, &errorOutput);
                     if (success) {
                         const QByteArray bytecode = readFile(tmpOut, FileType::Binary);
                         if (!bytecode.isEmpty()) {
-- 
2.44.0

