From 044fdabf54eb46b78d6dcd2e8f4386aa2bae2bad Mon Sep 17 00:00:00 2001
From: Fatih Uzunoglu <fuzun54@outlook.com>
Date: Fri, 19 Apr 2024 16:48:39 +0300
Subject: [PATCH] Set default compression level to 9 in RCC

---
 src/tools/rcc/rcc.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tools/rcc/rcc.cpp b/src/tools/rcc/rcc.cpp
index 4e5269cc99..c5562807c0 100644
--- a/src/tools/rcc/rcc.cpp
+++ b/src/tools/rcc/rcc.cpp
@@ -29,7 +29,7 @@ using namespace Qt::StringLiterals;
 
 enum {
     CONSTANT_USENAMESPACE = 1,
-    CONSTANT_COMPRESSLEVEL_DEFAULT = -1,
+    CONSTANT_COMPRESSLEVEL_DEFAULT = 9,
     CONSTANT_ZSTDCOMPRESSLEVEL_CHECK = 1,   // Zstd level to check if compressing is a good idea
     CONSTANT_ZSTDCOMPRESSLEVEL_STORE = 14,  // Zstd level to actually store the data
     CONSTANT_COMPRESSTHRESHOLD_DEFAULT = 70
-- 
2.44.0

